### PR TITLE
Add support for ViT for image classification

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

This PR adds support for ViT for image classification.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: tests/baselines/vit_base_patch16_224_in21k.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "beans": {
+        "num_train_epochs": 5,
+        "eval_batch_size": 8,
+        "distribution": {
+            "single_card": {
+                "learning_rate": 2e-5,
+                "train_batch_size": 8,
+                "eval_accuracy": 0.985,
+                "train_runtime": 92.8927,
+                "extra_arguments": [
```
Comment (libinta) [inline tests/baselines/vit_base_patch16_224_in21k.json:11]: do we have image per sec throughput measured?

File: examples/image-classification/README.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
+<!---
+Copyright 2021 The HuggingFace Team. All rights reserved.
```
[the remark this answers is not in the document]
Comment (regisss) [reply examples/image-classification/README.md:2]: The README is heavily inspired by the one of Transformers which has this copyright year, so I think we should keep it.

File: tests/baselines/vit_base_patch16_224_in21k.json (hunk 1 above)
Comment (regisss) [reply tests/baselines/vit_base_patch16_224_in21k.json:11]: I can measure it. So far we have not used it as a regression metric in CI, but we can add it in another PR.

**Review by libinta — APPROVED** [review]
